lib/plugins/chat.js

```diff
@@ -108,43 +108,54 @@ function inject (bot, options) {
 
   addDefaultPatterns()
 
-  // Pre 1.19
-  bot._client.on('chat', (packet) => {
-    const msg = ChatMessage.fromNotch(packet.message)
-    const chatPositions = {
-      0: 'chat',
-      1: 'system',
-      2: 'game_info'
-    }
-    bot.emit('message', msg, chatPositions[packet.position], packet.sender, /* verified */ null)
-    bot.emit('messagestr', msg.toString(), chatPositions[packet.position], msg, packet.sender, /* verified */ null)
-    // Position 2 is the action bar
-    if (packet.position === 2) bot.emit('actionBar', msg, null)
-  })
+  bot._client.on('playerChat', (data) => {
+    const message = data.formattedMessage
+    const verified = data.verified
+    let msg
+    if (bot.supportFeature('clientsideChatFormatting')) {
+      let content
 
-  // 1.19+
-  bot._client.on('player_chat', (packet) => {
-    const message = packet.unsignedChatContent || packet.signedChatContent
-    let verified = false
-    const sender = bot.uuidToUsername[packet.senderUuid]
-    if (sender) {
-      const { profileKeys } = bot.players[sender]
-      if (profileKeys) verified = bot._client.verifyMessage(profileKeys.publicKey, packet)
-    }
-    const parameters = {
-      sender: JSON.parse(packet.senderName),
-      content: JSON.parse(message)
+      if (message) content = JSON.parse(message)
+      else if (options.allowInsecureChat && data.unsignedContent) content = JSON.parse(data.unsignedContent)
+      else content = { text: data.plainMessage }
+
+      let sender
+
+      try {
+        sender = JSON.parse(data.senderName)
+      } catch {
+        sender = {
+          insertion: data.senderName,
+          clickEvent: { action: 'suggest_command', value: `/tell ${data.senderName} ` },
+          hoverEvent: { action: 'show_entity', contents: { id: bot.findPlayer(data.senderName).id, name: data.senderName } },
+          text: data.senderName
+        }
+      }
+
+      const parameters = {
+        sender,
+        content
+      }
+      msg = ChatMessage.fromNetwork(data.type, parameters)
+      Object.assign(msg, parameters)
+      if (data.unsignedContent) msg.unsignedContent = JSON.parse(data.unsignedContent)
+      msg.plainMessage = data.plainMessage
```

Comment:
`msg.plainMessage` shouldn't be necessary, that should be the primary message

Reply:
msg.plainMessage will not be the primary message when formattedMessage is defined. Regardless of whether we make the changes you requested or not

Reply:
Yeah, we can set the plainMessage to always be the default

Reply:
What do you mean?

Reply:
It makes more sense to expose it.